### ISSUE TITLE
 Permissive firewall policy detected #201

### ISSUE THREAD

**Issue by fshmcallister** [WoTTsecurity/wott-io#203]
We found permissive firewall policy present on {devices}. Please consider change it to more restrictive one.

as part of #198

**Comment by fshmcallister** [comment]
Title: Permissive firewall policy detected
> change to a more restrictive policy

**TL;DR**

Permissive firewall policy detected on [DEVICE/NODE/ENDPOINT(S)]. Permissive firewall policies permit all traffic unless otherwise specified. Note that you can have different firewall policies for ingoing and outgoing traffic, so check both. Consider changing your policy to be more restrictive, either by implementing rules to your permissive policy, or setting it to restrictive.

**FAQ**

Firewall policies typically come in two modes: permissive (sometimes known as passive) or restrictive. Permissive policies permit all traffic unless otherwise specified, but these rules can be complex to implement depending on the firewall provider. Restrictive policies essentially do the opposite, they deny all traffic unless otherwise specified. This is the most secure option as it requires manual editing of the access control list. 

In order to be as secure as possible, configure your policy to be as specific as possible. Your source IP address, target IP address, and destination port for traffic should be specified and the range of such should be minimal. Avoid using `any` where possible, the only scenario where this can't be the case is if you have a service that should be available to all internet users. So you may want to configure access to your `HTTPS` address as `any`, but ensure access to other backend services such as a database is limited. 

Alternatively, implement more rules to your permissive policy (although this is still not recommended). Both ingoing and outgoing traffic can be configured to go through different firewall policies. Make sure to check both for security weaknesses.

**Comment by AlEsmail** [comment]
Good job. Wait for Viktor to confirm ok.

**Comment by vpetersson** [comment]
> Note that you can have different firewall policies for ingoing and outgoing traffic

"Note that you can have different firewall policies for incoming (ingress) and outgoing traffic (egress)."

**Comment by vpetersson** [comment]
Good @fshmcallister

**Comment by fshmcallister** [comment]
updated original post to include ingress and egress

**Comment by fshmcallister** [comment]
**TITLE**
Permissive firewall policy detected
> change to a more restrictive policy

**TL;DR**

Permissive firewall policy detected. Permissive firewall policies permit all traffic unless otherwise specified. Note that you can have different firewall policies for ingoing (ingress) and outgoing (egress) traffic, so check both. Consider changing your policy to be more restrictive, either by implementing rules to your permissive policy, or setting it to restrictive.

**FAQ**

Firewall policies typically come in two modes: permissive (sometimes known as passive) or restrictive. Permissive policies permit all traffic unless otherwise specified, but these rules can be complex to implement depending on the firewall provider. Restrictive policies essentially do the opposite, they deny all traffic unless otherwise specified. This is the most secure option as it requires manual editing of the access control list. 

In order to be as secure as possible, configure your policy to be as specific as possible. Your source IP address, target IP address, and destination port for traffic should be specified and the range of such should be minimal. Avoid using \`any\` where possible, the only scenario where this can't be the case is if you have a service that should be available to all internet users. So you may want to configure access to your \`HTTPS\` address as \`any\`, but ensure access to other backend services such as a database is limited. 

Alternatively, implement more rules to your permissive policy (although this is still not recommended). Both ingress and egress traffic can be configured to go through different firewall policies. Make sure to check both for security weaknesses.
